diff --git a/src/Form/ISFieldSettingsForm.php b/src/Form/ISFieldSettingsForm.php
index 4a2b7fb..b602cf5 100644
--- a/src/Form/ISFieldSettingsForm.php
+++ b/src/Form/ISFieldSettingsForm.php
@@ -25,8 +25,15 @@ class ISFieldSettingsForm extends ConfigFormBase {
   /**
    * {@inheritdoc}
    */
+  protected function getEditableConfigNames() {
+    return ['isfield.settings'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
   public function buildForm(array $form, FormStateInterface $form_state) {
-    $config = \Drupal::config('isfield.settings');
+    $config = $this->config('isfield.settings');
     $form['video_dimensions'] = array(
       '#title' => t('Predefined video dimensions'),
       '#type' => 'textarea',
@@ -45,7 +52,7 @@ class ISFieldSettingsForm extends ConfigFormBase {
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
-    \Drupal::config('isfield.settings')
+    $this->config('isfield.settings')
       ->set('video_dimensions', explode("\n", str_replace("\r", "", $form_state->getValue('video_dimensions'))))
       ->set('thumbnail_dimensions', explode("\n", str_replace("\r", "", $form_state->getValue('thumbnail_dimensions'))))
       ->save();
